src/Expr.h

```diff
@@ -1502,6 +1508,8 @@ class ListExpr : public Expr {
     ExprPtr Reduce(Reducer* c, StmtPtr& red_stmt) override;
     StmtPtr ReduceToSingletons(Reducer* c) override;
 
+    bool AllConstantOps() const;
```

Comment:
Alternative to a new name might be to make HasConstantOps() virtual and override for ListExpr?

Reply:
Yeah, that's a better name. In principle it could be virtual and overridden for `ListExpr`, but to do that in a non-surprising way would entail implementing it for a bunch of other `Expr` subclasses too, which I'd rather avoid at this point given it's not needed in any other context.